cmd/avalanche-network-runner/server/server.go

```diff
@@ -80,10 +80,13 @@ func serverFunc(cmd *cobra.Command, args []string) (err error) {
 	case sig := <-sigc:
 		zap.L().Warn("signal received; closing server", zap.String("signal", sig.String()))
 		rootCancel()
-		zap.L().Warn("closed server", zap.Error(<-errc))
-	case err = <-errc:
-		zap.L().Warn("server closed", zap.Error(err))
-		rootCancel()
+		// wait for server stop
+		serverRunErr := <-errc
+		zap.L().Warn("closed server", zap.Error(serverRunErr))
+    case serverRunErr := <-errc:
```

Comment:
Can we name these differently? `serverRunErr` in both cases have nothing to do with each other, it might improve readability if named differently